packages/eslint-plugin/docs/rules/no-unnecessary-type-parameters.md

```diff
@@ -0,0 +1,104 @@
+---
+description: 'Disallow type parameters that only appear once'
+---
+
+> 🛑 This file is source code, not the primary documentation location! 🛑
+>
+> See **https://typescript-eslint.io/rules/no-unnecessary-type-parameters** for documentation.
+
+This rule forbids type parameters that only appear once in a function, method, or class definition.
+
+Type parameters relate two types. If a generic type parameter only appears once, then it is not relating anything. It can usually be replaced with explicit types such as `unknown`.
+
+At best unnecessary type parameters make code harder to read. At worst they can be used to disguise unsafe type assertions.
+
+## Options
+
+None.
+
```

Comment:
[Docs] Our tooling will auto-generate this ✨  https://github.com/typescript-eslint/typescript-eslint/blob/b1c92bbca17021605950308987ebc0383037f649/packages/website/plugins/generated-rule-docs/insertions/insertNewRuleReferences.ts#L74

```suggestion
```